File: src/pocketcloud/network/packet/impl/normal/ServerSyncPacket.php
<?php

namespace pocketcloud\network\packet\impl\normal;

use pocketcloud\network\client\ServerClient;
use pocketcloud\network\packet\CloudPacket;
use pocketcloud\network\packet\utils\PacketData;
use pocketcloud\server\CloudServer;

class ServerSyncPacket extends CloudPacket {

    public function __construct(
        private ?CloudServer $server = null,
        private bool $removal = false
    ) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeServer($this->server);
        $packetData->write($this->removal);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->server = $packetData->readServer();
        $this->removal = $packetData->readBool();
    }

    public function getServer(): ?CloudServer {
        return $this->server;
    }

    public function isRemoval(): bool {
        return $this->removal;
    }

    public function handle(ServerClient $client): void {}
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/ServerSyncPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.model.server.ICloudServer;
import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.network.packet.CloudPacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class ServerSyncPacket extends CloudPacket implements ClientboundPacket {

    private ICloudServer server;
    private boolean removal;

    public ServerSyncPacket(ICloudServer server, boolean removal) {
        this.server = server;
        this.removal = removal;
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(server, removal);
    }

    @Override
    public void decodePayload(IPacketData packetData) {}

    public static ServerSyncPacket create(ICloudServer server, boolean removal) {
        return new ServerSyncPacket(server, removal);
    }
}
